### PR TITLE
[FLOC-1824] Use Nova API calls to get compute_instance_id

### PR DESCRIPTION
**Pull request by wallrj**
Design for https://clusterhq.atlassian.net/browse/FLOC-1824

### CODE REVIEW

File: flocker/node/agents/functional/test_cinder.py [hunk 1]
```diff
@@ -107,3 +107,8 @@ def test_foreign_cluster_volume(self):
         self.addCleanup(blockdevice_api2.destroy_volume,
                         flocker_volume.blockdevice_id)
         self.assert_foreign_volume(flocker_volume)
+
+    # Maybe add an implementation specific test here to verify that
+    # the ID returned by compute_instance_id() can be retreived using
+    # the nova server get API and that it has our IP addesses
+    # http://developer.openstack.org/api-ref-compute-v2.html#getServer
```
Comment (exarkun) [inline flocker/node/agents/functional/test_cinder.py]: I think our existing test for `get_device_path` might cover this.  It requires that after attaching to our `compute_instance_id` the reported device path is an actual block device on the system where the tests are running.  That's a pretty good indication that we've gotten the right compute instance id.

File: flocker/node/test/test_script.py [hunk 2]
```diff
@@ -90,6 +90,8 @@ def get_all_ips():
     """
     Find all IPs for this machine.
 
+    XXX: Move this to flocker.common.???
+
```
Comment (exarkun) [inline flocker/node/test/test_script.py]: Hm.  This is really `get_all_ipv4s`.  Maybe put a real `get_all_ips` (ipv4 and ipv6) in `flocker.common` and we can think about transitioning everything to that and away from this.

You're planning to include ipv6 addresses in the matching, right?

**Comment by exarkun** [comment]
Thanks.  Looks pretty good.  One other thing to think about ... is `xenstore-read` better than IP matching on systems where `xenstore-read` exists?

File: flocker/node/agents/functional/test_cinder.py (hunk 1 above)
Comment (wallrj) [inline flocker/node/agents/functional/test_cinder.py]: True. This isn't needed then.

File: flocker/node/test/test_script.py (hunk 2 above)
Comment (wallrj) [inline flocker/node/test/test_script.py]: Yep, I'll modify it to get *all* IPs...assuming the API returns IPv6 addresses.

**Comment by wallrj** [comment]
`xenstore-read` is faster  and more precise but requires root.
`/var/lib/cloud/data/instance-id` is as precise, but available to non-root users.

We could implement a function that tried these in order, falling back to API calls but, as discussed, it'll add complexity and I'll leave that for a follow up issue.

**Pull request closed without merging**